### PR TITLE
feat(moplat-233):shows

### PR DESCRIPTION
**Pull request by patrinoua**
This PR resolves [moplat-233]

### Description

- add tests

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

### CODE REVIEW

File: src/Screens/HomeTabs/Shows/Shows.tsx [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
+            totalCount
+            edges {
+              node {
+                name
+                startAt
+                endAt
+                formattedStartAt: startAt(format: "MMMM D")
+                formattedEndAt: endAt(format: "MMMM D, YYYY")
+                coverImage {
+                  url
```
Comment (patrinoua) [inline src/Screens/HomeTabs/Shows/Shows.tsx:24]: What is the right field here, url, imageURL? @pvinis any idea?

File: src/Screens/HomeTabs/Shows/Shows.tsx [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+                }
+              }
+            }
+          }
+        }
+      }
+    `, //@ts-ignore
+    { partnerID }
```
Comment (rajsam003) [inline src/Screens/HomeTabs/Shows/Shows.tsx:32]: Lets avoid adding @ts-ignore

```suggestion
    { partnerID: partnerID! }
```

File: src/Screens/HomeTabs/Shows/Shows.tsx [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
 export const Shows = () => {
-  return (
-    <Tabs.ScrollView>
-      <Text>Shows</Text>
-      <Text>Shows</Text>
-      <Text>Shows</Text>
-    </Tabs.ScrollView>
+  const partnerID = GlobalStore.useAppState((state) => state.activePartnerID)
+  const data = useLazyLoadQuery<ShowsTabQuery>(
+    graphql`
```
Comment (pvinis) [inline src/Screens/HomeTabs/Shows/Shows.tsx:11]: one thing we could do here is move that query in a variable below the component, so we keep the component code a tiny bit cleaner, like

```
export const Shows = () => {
  const partnerID = GlobalStore.useAppState((state) => state.activePartnerID)
  const data = useLazyLoadQuery<ShowsTabQuery>(showsQuery)
..
}

const showsQuery = graphql`
  query ShowsTabQuery
  ..
`
```

File: src/Screens/HomeTabs/Shows/Shows.tsx (hunk 1 above)
Comment (pvinis) [reply src/Screens/HomeTabs/Shows/Shows.tsx:24]: not sure. i would check on eigen maybe, or ask in #dev-help 🤔.

File: src/Screens/HomeTabs/Shows/ShowsTabFlatList/ShowsTabFlatList.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+export const ShowsTabFlatList = ({ shows }: { shows: any }) => {
+  return (
+    <TabsFlatList
+      data={shows}
+      renderItem={({ item: show }) => <ShowOverview show={show} />}
+      // keyExtractor={(item) => item?.internalID!}
+    />
+  )
+}
+//remove any before merging
```
Comment (pvinis) [inline src/Screens/HomeTabs/Shows/ShowsTabFlatList/ShowsTabFlatList.tsx:16]: ;)

File: src/Screens/HomeTabs/Shows/Shows.tsx (hunk 1 above)
Comment (pvinis) [reply src/Screens/HomeTabs/Shows/Shows.tsx:22]: i think we figured this out right?

File: src/Screens/HomeTabs/Shows/ShowsTabFlatList/ShowsTabFlatList.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import React from "react"
+import { Flex, Text } from "palette"
+import { Image } from "react-native"
+import { TabsFlatList } from "helpers/components/TabsWrapper"
+
+//remove any before merging
+export const ShowsTabFlatList = ({ shows }: { shows: any }) => {
```
Comment (pvinis) [inline src/Screens/HomeTabs/Shows/ShowsTabFlatList/ShowsTabFlatList.tsx:7]: do we need a component just to return the TabsFlatList? why not just using that directly in the Shows.tsx file? 🤔

File: src/Screens/HomeTabs/Shows/Shows.tests.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  it("renders without throwing an error", async () => {
+    renderWithWrappersTL(<Shows />)
+    await mockEnvironmentPayloadMaybe(mockProps)
+  })
+
+  it("renders the list of shows", async () => {
+    const { queryAllByText } = renderWithWrappersTL(<Shows />)
+    await mockEnvironmentPayloadMaybe(mockProps)
+
+    expect(queryAllByText("Gustav Klimts shows")).toHaveLength(10)
```
Comment (pvinis) [inline src/Screens/HomeTabs/Shows/Shows.tests.tsx:16]: nice

File: src/Screens/HomeTabs/Shows/Shows.tsx (hunk 3 above)
Comment (pvinis) [reply src/Screens/HomeTabs/Shows/Shows.tsx:11]: i think it makes sense to put the query outside the component in all cases. it seems a bit cleaner? but yea, lets talk tomorrow?

**Review by pvinis — CHANGES_REQUESTED** [review]
i left some comments. let me know what you think.

**Comment by patrinoua** [comment]
replaced from https://github.com/artsy/energy/pull/48

**Pull request closed without merging**